### ISSUE TITLE
"azure appservice [plan|web]" commands return no data

### ISSUE THREAD

**Issue by mikebridge** [Azure/azure-cli#2841]
### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here: pip (python 3.6.0)

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here: 2.0.2 

**OS Version:** What OS and version are you using?  
Answer here: Windows 10

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here: cmd.exe

---

### Description

The command `az appservice plan list` returns an empty json array, even though I have App Services visible in the portal.  Similar problem with `az appservice web list`.  `az appservice plan show -g MyGroup -n myname` returns nothing at all, but typing an invalid name returns an error.

I successfully created a plan via the command line, but it does not appear in the list (though it appears in the portal):

```
> az appservice plan create -n TestCLI -g MyResourceGroup
 => {
  "adminSiteName": null,
  "appServicePlanName": "TestCLI",
  "geoRegion": "South Central US",
  "hostingEnvironmentProfile": null,
  ... 
}
> az appservice plan list -g MyResourceGroup
=> []
```

However it *is* listed in `azure appserviceplan list --resource-group MyResourceGroup`

Also, `az webservice web list` returns nothing.  Not sure if that's related.

**Comment by yugangw-msft** [comment]
The difference between CLI 1.0 and 2.0 is the api-version. CLI 1.0, `azure`, uses an older version. Do you mind running them in verbose mode and send the output to me `yugangw at microsoft.com`
```bash
azure appserviceplan list --resource-group MyResourceGroup -vv
az appservice plan list -g MyResourceGroup --debug
```

**Comment by mikebridge** [comment]
I sent you the output, thanks!

BTW, the az commands also do not work on Linux with python 2.7.

**Comment by yugangw-msft** [comment]
> BTW, the az commands also do not work for me on Linux with python 2.7.

I am using 2.7 on linux all the time and it works. Could you share a bit error details?

**Comment by mikebridge** [comment]
@yugangw-msft the az output looks to be the same on linux.

**Comment by yugangw-msft** [comment]
@mikebridge, got it. I misunderstood you meant a more generic issue on linux. And yep, this is expected for the same command failing in linux as this is a service side issue, which I will forward to webapp team.

**Comment by markjbrown** [comment]
I have the same issue. Repro script here below.

az group create --name GroupCentralUS --location centralus
gitrepo=https://github.com/Azure-Samples/app-service-web-dotnet-get-started.git
az appservice plan create --name MyWebAppCentralUS --resource-group GroupCentralUS --sku FREE
az appservice web create --name MyWebAppCentralUS --resource-group GroupCentralUS --plan MyWebAppCentralUS
az appservice web source-control config --name MyWebAppCentralUS --resource-group GroupCentralUS --repo-url $gitrepo --branch master --manual-integration

# show the website (returns nothing)
az appservice web list --resource-group GroupCentralUS

**Comment by yugangw-msft** [comment]
@markjbrown , I tried your repro, still no repro :-(

**Comment by yugangw-msft** [comment]
fixed by #2758
